docs/source/federation-spec.mdx

```diff
@@ -0,0 +1,625 @@
+
+### Resolving `Query._entities`
+
+As a reminder, here's the definition of the `Query._entities` field that every subgraph must automatically define (unless that subgraph defines zero entities):
+
+```graphql {2}
+type Query {
+  _entities(representations: [_Any!]!): [_Entity]!
+}
+```
+
+Every subgraph must also automatically define the resolver for this field. The  logic for this resolver is as follows:
+
+1. For each entity representation included in the `representations` list, obtain the `__typename` from that representation.
+2. Execute the **reference resolver** for the corresponding `__typename` and provide the full representation to that resolver.
+3. Include the reference resolver's result in the returned `_Entity` list.
+
+For example, the `@apollo/subgraph` library defines the following resolver for `Query._entities`:
+
+```js
+{
+  _entities: (_source, { representations }, context, info) => entitiesResolver({ representations, context, info })
+}
+```
+
+This passes the list of representations to an `entitiesResolver` helper function, which is [defined here](https://github.com/apollographql/federation/blob/main/subgraph-js/src/types.ts#L64-L115).
+
+## Glossary of schema additions
+
+This section describes all of the types and field definitions that a valid subgraph service must automatically add to its schema. These definitions are all listed in [Subgraph schema specification](#subgraph-schema-specification).
+
+### == Added `Query` fields ==
+
+### `Query._service`
+
+This field of the root `Query` type must return a non-nullable [`_Service` type](#type-_service).
+
+For details, see [Enhanced introspection with `Query._service`](#enhanced-introspection-with-query_service).
+
+### `Query._entities`
+
+The graph router uses this root-level `Query` field to directly fetch fields of entities defined by a subgraph.
+
+This field must take a `representations` argument of type `[_Any!]!` (a non-nullable list of non-nullable [`_Any` scalars](#scalar-_any)). Its return type must be `[_Entity]!` (a non-nullable list of _nullable_ objects that belong to the [`_Entity` union](#union-_entity)).
+
+Each entry in the `representations` list  must be validated with the following rules:
+
+* A representation must include a `__typename` string field.
+* A representation must contain all fields included in the fieldset of a `@key` directive applied to the corresponding entity definition.
+
+For details, see [Resolving entity fields with `Query._entities`](#resolving-entity-fields-with-query_entities).
+
+### == Added types ==
+
+### `type _Service`
+
+This object type must have an `sdl: String!` field, which returns the SDL of the subgraph schema as a string.
+
+The returned schema string _must_ include all uses of federation-specific directives (`@key`, `@requires`, etc.), and it _must not_ include any definitions from the [subgraph schema specification](#subgraph-schema-specification).
```

Comment:
again this is true for fed v1, fed v2 can include full SDL definition with all the types